storage/src/vespa/storage/persistence/persistencethread.cpp

```diff
@@ -950,22 +949,15 @@ PersistenceThread::run(framework::ThreadHandle& thread)
         if (lock.first) {
             processLockedMessage(std::move(lock));
         }
-
-        vespalib::MonitorGuard flushMonitorGuard(_flushMonitor);
-        flushMonitorGuard.broadcast();
     }
     LOG(debug, "Closing down persistence thread");
-    vespalib::MonitorGuard flushMonitorGuard(_flushMonitor);
-    _closed = true;
-    flushMonitorGuard.broadcast();
 }
 
 void
 PersistenceThread::flush()
 {
-    vespalib::MonitorGuard flushMonitorGuard(_flushMonitor);
-    if (!_closed) {
-        flushMonitorGuard.wait();
+    while (_env._fileStorHandler.getQueueSize() != 0) {
```

Comment:
Could potentially just check the queue size for the thread's stripe, but `getQueueSize()` is a superset of that stripe so shouldn't matter in practice.

Reply:
I will add an api for that after disk concept is removed.